lib/kill.js

```diff
@@ -65,6 +81,11 @@ export const validateTimeout = ({timeout}) => {
 	}
 };
 
+// Indicates that the error is used only to interrupt control flow, but not in the return value
+const setDiscardedError = error => Object.assign(error, {[discardSymbol]: true});
+export const isDiscardedError = error => error?.[discardSymbol] === true;
+const discardSymbol = Symbol('discard');
```

Comment:
For timeouts, non-0 exit codes and signals, we do not want to add anything to the error message. We are also not interested in the stack trace, since the error would be created inside the Execa code base. 

Instead, we rely on the main `makeError()` logic to set the right error message, and create some stack trace.